File: app/src/main/java/com/karthik/myweather/ui/viewModel/BaseViewModel.java
package com.karthik.myweather.ui.viewModel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class BaseViewModel extends ViewModel {

    public static final String Error = "Error";

    public MutableLiveData<Boolean> isLoading = new MutableLiveData<>();
    public MutableLiveData<String> error = new MutableLiveData<>();
}

File: app/src/main/java/com/karthik/myweather/ui/viewModel/BaseViewModel.kt
package com.karthik.myweather.ui.viewModel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

open class BaseViewModel : ViewModel() {
    @JvmField
    val isLoading = MutableLiveData<Boolean>()

    @JvmField
    val error = MutableLiveData<String>()

    companion object {
        const val Error = "Error"
    }
}